prefs.js

```diff
@@ -109,6 +109,36 @@ var SettingsWidget = class SettingsWidget {
             this._settings.set_int('vertical-margin-bottom', bottomMargin.get_value());
         });
 
+        let cycleWidths = this.builder.get_object('cycle_widths_entry');
+        let widthArray = this._settings.get_value('cycle-width-steps').deep_unpack();
+        cycleWidths.set_text(widthArray.toString().replaceAll(',', ', '));
+        cycleWidths.connect('changed', () => {
+            let varr = cycleWidths.get_text()
+                .replaceAll(',', ' ')
+                .split(/\s+/)
```

Comment:
> can be entered as comma (or whitespace) separated values

I would be stricter with how you parse this.

1. Require a certain delimiter, rather than allowing the user to mix whitespace and commas. Fine to also allow the other one to be present optionally (`, ` separators are easier for *me* to read so I'd like if they were *supported*), but I would require one or the other as the character we actually split on. I don't think there's any edge cases where this matters right now; it's just a case of "why make parsing harder for ourselves if it's not strictly necessary?"
    - I recommend making the whitespace the optional bit. I don't think it will be relevant here, but making whitespace not be a separator leaves open the _option_ of doing crazy things like `max 50% 1000px` as a value. Not saying we _should_ add anything along those lines, just that it's nice to have the flexibility if needed.

2. I'd use a semicolon as a separator, rather than a comma. In much of Europe, comma is a decimal place separator, so it's something we might someday want to support (`33,33%`).

```suggestion
        cycleWidths.connect('changed', () => {
            let varr = cycleWidths.get_text()
                .split(';')
                .map((s) => s.trim())
```

Note: this is (obviously) not the only change that would be needed, just a convenient place to hang this thread.

Reply:
Yep, makes sense.  The original implementation was quite lenient on input format since I only wanted some delimited number array (hence replacing and splitting on whitespace chunks and then mapping to Number worked well).  Given we'll be switching to `%` and `px` strings trimming and processing seems in order.

Reply:
Okay!  requesting a re-review here.  Have implemented all these suggestions.  Parsing is stricter and am enforcing units on each input.  Thanks @smichel17!